configure.ac

```diff
@@ -665,14 +672,15 @@ AS_CASE([$host],
   [*-*-cygwin*],
     [mksharedlib="$flexlink"
     mkmaindll="$flexlink -maindll"
-    shared_libraries_supported=true],
+    shared_libraries_supported=$with_sharedlibs],
```

Comment:
The change here is not mandatory, but it propagated `--disable-shared` in a better way - it means that `win32.c` is correctly compiled without dynamic linking support but then the build still fails later when flexlink isn't found (that's a much a bigger change to make)